### PR TITLE
feat: Build the structure of the tool [TAROT-2350]

### PR DESCRIPTION
**Pull request by Afonsobli**

### CODE REVIEW

File: .codacyrc
Comment (mrfyda) [inline .codacyrc:1]: used for testing only. use with:

```bash
docker run -it -v `pwd`/docs/multiple-tests/suggestions/src:/src -v `pwd`/.codacyrc:/.codacyrc codacy-semgrep
```

File: internal/tool/tool.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+var _ codacy.Tool = (*codacySemgrep)(nil)
+
+// Run runs the Semgrep implementation
+func (s codacySemgrep) Run(ctx context.Context, toolExecution codacy.ToolExecution) ([]codacy.Result, error) {
+	configFile, err := getConfigurationFile(*toolExecution.Patterns, toolExecution.SourceDir)
+	if err == nil {
+		defer os.Remove(configFile.Name())
+	}
+
+	// filesToAnalyse, err := getListOfFilesToAnalyse(tool.Files, sourceDir)
```
Comment (mrfyda) [inline internal/tool/tool.go:39]: not sure why this is needed 🤷 
perhaps this logic predates the golang seed

File: internal/tool/tool.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+		}
+	}
+	return []string{}
+}
+
+func commandParameters(configFile *os.File, filesToAnalyse []string) []string {
+	cmdParams := append(
+		[]string{
+			"-json", "-json_nodots",
+			"-lang", "python", // TODO: get language from toolExecution?
```
Comment (mrfyda) [inline internal/tool/tool.go:130]: Codacy will never invoke the tool for multiple languages at once. it shouldn't be a problem if we lookup the language from the pattern information in the tool definition

File: internal/tool/tool.go [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+	if len(patterns) == 0 {
+		sourceConfigFileContent, err := configurationFromSourceCode(sourceFolder)
+		if err == nil {
+			return writeToTempFile(sourceConfigFileContent)
+		}
+
+		return nil, err
+	}
+
+	// TODO: generate configuration file from patterns and auto config file
```
Comment (mrfyda) [inline internal/tool/tool.go:109]: here we need to generate a file with the rule definitions for the enabled patterns. we can use the 'auto' configuration file from the registry and extract the enabled rules to a temporary file

File: internal/tool/tool.go (hunk 1 above)
Comment (mrfyda) [reply internal/tool/tool.go:39]: see https://codacy.atlassian.net/browse/TAROT-2433 why it's need
can be addressed in a follow up PR

File: internal/tool/tool.go (hunk 2 above)
Comment (Afonsobli) [reply internal/tool/tool.go:130]: We will need to find a way to check for the language as that definition is not present in the tool definition

**Pull request merged**